kedro/contrib/io/matplotlib/matplotlib_writer.py

```diff
@@ -0,0 +1,123 @@
+
+        Example:
+        ::
+
+            >>> import matplotlib.pyplot as plt
+            >>> from kedro.contrib.io.matplotlib import MatplotlibWriter
+            >>>
+            >>> plt.plot([1,2,3],[4,5,6])
+            >>>
+            >>> single_plot_writer = MatplotlibWriter(filepath="data/new_plot.png")
+            >>> single_plot_writer.save(plt)
+            >>>
+            >>> plt.close()
+            >>>
+            >>> plots_dict = dict()
+            >>>
+            >>> for colour in ['blue', 'green', 'red']:
+            >>>     plots_dict[colour] = plt.figure()
+            >>>     plt.plot([1,2,3],[4,5,6], color=colour)
+            >>>     plt.close()
+            >>>
+            >>> dict_plot_writer = MatplotlibWriter(filepath="data/")
+            >>> dict_plot_writer.save(plots_dict)
+            >>>
+            >>> plots_list = []
+            >>> for index in range(5):
+            >>>    plots_list.append(plt.figure())
+            >>>    plt.plot([1,2,3],[4,5,6], color=colour)
+            >>> list_plot_writer = MatplotlibWriter(filepath="data/")
+            >>> list_plot_writer.save(plots_list)
+
+    """
+
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibWriter``.
+
+        Args:
+            filepath: Path to a matplot object file.
+            load_args: Currently ignored as loading is not supported.
+            save_args: Save args passed to `plt.savefig`. See
+                https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html
+        """
+        self._filepath = Path(filepath)
+        self._load_args = load_args if load_args else dict()
+        self._save_args = save_args if save_args else dict()
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+    def _load(self) -> None:
+        raise DataSetError(
+            "Loading not supported for `{}`".format(self.__class__.__name__)
+        )
+
+    def _save(self, data: Union[figure, List[figure], Dict[str, figure]]) -> None:
+        if isinstance(data, list):
+            self._filepath.mkdir(exist_ok=True)
+            for index, plot in enumerate(data):
+                plot.savefig(str(self._filepath / str(index)), **self._save_args)
+
+        elif isinstance(data, dict):
+            self._filepath.mkdir(exist_ok=True)
+            for plot_name, plot in data.items():
+                plot.savefig(str(self._filepath / plot_name), **self._save_args)
+        else:
+            data.savefig(str(self._filepath), **self._save_args)
+
+    def _exists(self) -> bool:
+        return self._filepath.is_file() or (
+            self._filepath.is_dir() and len(list(self._filepath.iterdir())) != 0
```

Comment:
This can be further simplified to be:
```suggestion
            self._filepath.is_dir() and list(self._filepath.iterdir())
```

Reply:
Seems like I have to use `len` because mypy doesn't like it and returns. 
` Incompatible return value type (got "Union[bool, List[Path]]", expected "bool")`